homeassistant/components/pilight/light.py

```diff
@@ -61,7 +61,18 @@ def supported_features(self):
 
     def turn_on(self, **kwargs):
         """Turn the switch on by calling pilight.send service with on code."""
-        self._brightness = kwargs.get(ATTR_BRIGHTNESS, 255)
-        dimlevel = int(self._brightness / (255 / self._dimlevel_max))
+        # Update brightness only if provided as an argument, this will allow the switch to keep its previous brightness level.
+        if ATTR_BRIGHTNESS in kwargs:
+            self._brightness = kwargs[ATTR_BRIGHTNESS]
+
+        dimlevel = None
+        if self._brightness is not None:
+            # Calculate pilight brightness (as a range of 0 to 15)
+            # By creating a percentage
+            percentage = self._brightness / 255
+            # Then calculate the dimmer range (aka amount of available brightness steps).
+            dimrange = self._dimlevel_max - self._dimlevel_min
+            # Finally calculate the pilight brightness, adding dimlevel_min back in to ensure the minimum is always reaced.
+            dimlevel = int(percentage * dimrange + self._dimlevel_min)
```

Comment:
It seems like you are still calculating `dimlevel` even when `ATTR_BRIGHTNESS` is not provided as an argument.

```suggestion
        # Update brightness only if provided as an argument, this will allow the switch to keep its previous brightness level.
        dimlevel = None

        if ATTR_BRIGHTNESS in kwargs:
            self._brightness = kwargs[ATTR_BRIGHTNESS]

            # Calculate pilight brightness (as a range of 0 to 15)
            # By creating a percentage
            percentage = self._brightness / 255
            # Then calculate the dimmer range (aka amount of available brightness steps).
            dimrange = self._dimlevel_max - self._dimlevel_min
            # Finally calculate the pilight brightness, adding dimlevel_min back in to ensure the minimum is always reaced.
            dimlevel = int(percentage * dimrange + self._dimlevel_min)
```

Reply:
Thanks for your suggestion :+1:
I wan't sure before if I had to reuse any lingering values for `self._brightness`. 
Looking at the other light integrations I agree this change is not only correct, but also easier to understand.